backend/app/controllers/user/getOrCreateUser.js

```diff
@@ -0,0 +1,33 @@
+const User = require( '../../../schema/user' );
+const DBConnection = require( '../../database/DBConnection' );
+const GoogleOAuthClient = require( '../../auth/GoogleOAuthClient' );
+
+module.exports = async ( req, res ) => {
+   try {
+      const token = req.header( 'X-GOOGLE-AUTH' );
+
+      if ( !token ) {
+         throw new Error( 'Missing user token' );
+      }
+
+      const OAuthClient = new GoogleOAuthClient();
+      const googleUser = await OAuthClient.verify( token );
+
+      await new DBConnection().make();
+      let user = await User.findOne( { googleId: googleUser.id } ).exec();
+
+      if ( !user ) {
+         const newUser = {
+            name: googleUser.name,
+            email: googleUser.email,
+            googleId: googleUser.id,
+         };
+         user = await new User( newUser ).save();
+      }
+
+      res.send( user );
+   }
+   catch ( e ) {
+      res.status( 400 ).send( e.message );
+   }
+};
```

Comment:
You included code on several abstraction levels in one function. This function handles an http request, gets google user's info and opens db connection and sends queries to the db. It's better to have multiple functions where each function does one thing than one function that does everything.

Reply:
Fixed by #85 .